compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -462,14 +462,25 @@ fn encode_ty<'tcx>(
                 ty::Uint(UintTy::U64) => "u3u64",
                 ty::Uint(UintTy::U128) => "u4u128",
                 ty::Uint(UintTy::Usize) => "u5usize",
-                ty::Float(FloatTy::F32) => "u3f32",
-                ty::Float(FloatTy::F64) => "u3f64",
                 _ => "",
             });
             compress(dict, DictKey::Ty(ty, TyQ::None), &mut s);
             typeid.push_str(&s);
         }
 
+        // Rust's f32 and f64 single (32-bit) and double (64-bit) precision floating-point types
+        // have IEEE-754 binary32 and binary64 floating-point layouts, respectively.
+        //
+        // (See https://rust-lang.github.io/unsafe-code-guidelines/layout/scalars.html#fixed-width-floating-point-types.)
+        ty::Float(..) => {
+            let s = String::from(match ty.kind() {
+                ty::Float(FloatTy::F32) => "f",
+                ty::Float(FloatTy::F64) => "d",
+                _ => "",
```

Comment:
Please remove this arm, this is essentially asking to introduce a bug here by making it possible to forget future float types in the future.

Reply:
Bind the float kind above on line 475 and then match that `FloatTy` here rather than calling `Ty::kind` again here.

Reply:
Done. Thanks!